File: Tree/104.Maximum_Depth_of_Binary_Tree/Maximum_Depth_of_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int mx = 0;
    void dfs(TreeNode *root, int curr){
        if(root != nullptr){
            curr += 1;
            mx = max(mx, curr);
            if(root->left != nullptr) dfs(root->left, curr);
            if(root->right != nullptr) dfs(root->right, curr);
        }
    }
    int maxDepth(TreeNode* root) {
        int curr = 0;
        dfs(root, curr);
        return mx;
    }
};

File: Tree/104.Maximum_Depth_of_Binary_Tree/Maximum_Depth_of_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.mx = 0
        
    def dfs(self, root, curr):
        if root:
            curr += 1
            self.mx = max(self.mx, curr)
            self.dfs(root.left, curr)
            self.dfs(root.right, curr)
            
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        curr = 0
        self.dfs(root, curr)
        return self.mx